sklearnex/linear_model/tests/test_incremental_linear.py

```diff
@@ -0,0 +1,202 @@
+# ===============================================================================
+# Copyright 2024 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# ===============================================================================
+
+import numpy as np
+import pytest
+from numpy.testing import assert_allclose
+
+from onedal.tests.utils._dataframes_support import (
+    _convert_to_dataframe,
+    get_dataframes_and_queues,
+)
+from sklearnex._device_offload import _transfer_to_host
```

Comment:
```suggestion
import sklearn.linear_model.tests.test_base
from sklearnex._device_offload import _transfer_to_host
```
I want to see some of sklearn's linear regression tests be applied to IncrementalLinearRegression via pytest monkeypatch. There are are lot of tests here, but I want to look specifically at mathematical verification.  Tests like test_linear_regression, test_fit_intercept, test_linear_regression_multiple_outcome

Reply:
all of them are not supported by c++ onedal side since there are not enough samples in their datasets

Reply:
Good to know, is this stated or checked anywhere on the python-side? I can't seem to find it. What is the value of the sample limit?

Reply:
https://github.com/olegkkruglov/scikit-learn-intelex/blob/d5233d8f71b6e827079f4b5e7fec994c6ffe2b74/sklearnex/linear_model/incremental_linear.py#L194

Reply:
and what do you mean by mathematical verification btw? i hope here
https://github.com/olegkkruglov/scikit-learn-intelex/blob/inc-lin/sklearnex/linear_model/tests/test_incremental_linear.py
and here
https://github.com/olegkkruglov/scikit-learn-intelex/blob/inc-lin/onedal/linear_model/tests/test_incremental_linear_regression.py
must be enough tests to check correctness of the algorithm

Reply:
@olegkkruglov is the "is_good_for_onedal" a carryover from the onedal LinearRegression implementation, or could you explain to me the limitations on onedal side for the algorithm to necessitate it? From what I understand, it was necessary in LinearRegression due to the use of QR decomposition , but what about in IncrementalLinearRegression?

Reply:
it is exactly the same algorithm with the same limitations